semana-17/aula-1_Autenticacao_Backend/src/services/cryptography/BcryptService.ts

```diff
@@ -0,0 +1,23 @@
+import { CryptographyGateway } from '../../business/gateways/crypt/CryptographyGateway';
+import * as bcrypt from "bcrypt";
+
+export class BcryptService implements CryptographyGateway {
+    
+    private static BCRYPT_SALT_ROUNDS: 10
+    
+    async encrypt(word: string): Promise<string> {
+        const salt = await bcrypt.genSalt(BcryptService.BCRYPT_SALT_ROUNDS);
+        const encryptWord = await bcrypt.hash(
+            word,
+            salt
+        )
+        return encryptWord;
+    }
+    async compare(word: string, hash: string): Promise<boolean> {
+        return await bcrypt.compare(word, hash)
```

Comment:
Uma dica sobre um pequeno detalhe. Na maioria das vezes, dar um await em um return não possui nenhum efeito, e, portanto, ele pode ser omitido. Essa pergunta no StackOverflow ilustra bem isso: https://stackoverflow.com/questions/38708550/difference-between-return-await-promise-and-return-promise